File: 8kyu/arrays/smallest-integer-array.js
/*
CHALLENGE DESCRIPTION

Given an array of integers your solution should find the smallest integer.

For example:

Given [34, 15, 88, 2] your solution will return 2
Given [34, -345, -1, 100] your solution will return -345
You can assume, for the purpose of this kata, that the supplied array will not be empty.
*/

// MY SOLUTION
function findSmallestInt(arr) {
    let min = Math.min(...arr);
    return min;
}

// OTHER SOLUTIONS
class SmallestIntegerFinder {
    findSmallestInt(args) {
      return Math.min(...args);
    }
}

class SmallestIntegerFinder {
    findSmallestInt(args) {
      return args.sort((a, b) => a - b)[0];
    }
}

class SmallestIntegerFinder {
    findSmallestInt(args) {
      return Math.min.apply(null, args);
    }
}

File: codewars/8kyu/arrays-and-fundamentals/smallest-int-arr/smallest_integer_array.py
""" 
CHALLENGE DESCRIPTION

Given an array of integers, your solution should find the smallest integer.

For example:
Given [34, 15, 88, 2], your solution should return 2.
Given [34, -345, -1, 100], your solution should return -345.

You can assume for the purpose of this kata that the supplied array will not be empty.
"""

# MY SOLUTION
def find_smallest_int(arr):
    smallest_num = float("inf")

    for num in arr:
        if num < smallest_num:
            smallest_num = num
    
    return smallest_num